File: Frontend/AI_socials/src/components/features/widgets/services/news.ts
import { fetchTopHeadlines } from "../api/gnews";
import type { NewsArticle } from "../types/NewsArticle";

function mapArticle(article: any): NewsArticle {
  return {
    id: article.articleUrl,
    title: article.title,
    description: article.description,
    imageUrl: article.image,
    articleUrl: article.url,
    source: article.source.name,
    publishedAt: article.publishedAt,
  };
}

export interface GetNewsOptions {
  category?: string;
  max?:number
}

export async function getNews(
  options: GetNewsOptions = {},
): Promise<NewsArticle[]> {
  const response = await fetchTopHeadlines(options);
  return response.articles.map(mapArticle);
}

File: Backend/agent_os/schemas/news.py
# here im going to define the schema for news_headlines
from datetime import datetime
from pydantic import BaseModel, HttpUrl


class Source(BaseModel):
    name: str
    url: HttpUrl


class NewsHeadline(BaseModel):
    title: str
    description: str | None = None
    category: str | None = None
    image_url: HttpUrl | None = None
    url: HttpUrl
    published_at: datetime
    source: Source
